### PR TITLE
Fix promote for /

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

**Comment by odow** [comment]
Does this fix https://github.com/jump-dev/MutableArithmetics.jl/issues/52?

File: src/interface.jl [hunk 1]
```diff
@@ -15,6 +15,9 @@ function promote_operation(op::Function, x::Type{<:AbstractArray}, y::Type{<:Abs
     # We replace it by a more helpful error here.
     error("`promote_operation($op, $x, $y)` not implemented yet, please report this.")
 end
+function promote_operation(::typeof(/), x::Type, y::Type)
```
Comment (odow) [inline src/interface.jl:18]: I learned of this recently: https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing

Not sure of the performance implications for this though.

Comment (blegat) [reply src/interface.jl:18]: I noticed this issue with `Vararg`, it's nice that it's documented now:
https://github.com/jump-dev/MutableArithmetics.jl/blob/c4e50e0056f523be07450aa791c8e60f8a2a0855/src/MutableArithmetics.jl#L9-L13

Comment (blegat) [reply src/interface.jl]: I agree in this case, it should always specialize anyway so let's force it

**Pull request merged**